tests/garage/torch/modules/test_cnn_module.py

```diff
@@ -0,0 +1,183 @@
+            paddings=paddings,
+            padding_mode='zeros',
+            hidden_nonlinearity=torch.relu,
+            hidden_w_init=nn.init.xavier_uniform_)
+
+        module_with_nonlinear_module_instance_and_function = CNNModule(
+            input_var=self.input,
+            hidden_channels=hidden_channels,
+            kernel_sizes=kernel_sizes,
+            strides=strides,
+            paddings=paddings,
+            padding_mode='zeros',
+            hidden_nonlinearity=nn.ReLU(),
+            hidden_w_init=nn.init.xavier_uniform_)
+
+        output1 = module_with_nonlinear_function_and_module(self.input)
+        output2 = module_with_nonlinear_module_instance_and_function(
+            self.input)
+
+        current_size = self.input_width
+        for (filter_size, stride, padding) in zip(kernel_sizes, strides,
+                                                  paddings):
+            # padding = float((filter_size - 1) / 2) # P = (F - 1) /2
+            current_size = int(
+                (current_size - filter_size + padding * 2) /
+                stride) + 1  # conv formula = ((W - F + 2P) / S) + 1
+        flatten_shape = current_size * current_size * hidden_channels[-1]
+
+        assert output1.shape == (self.batch_size, flatten_shape)
+        assert output2.shape == (self.batch_size, flatten_shape)
+
+    @pytest.mark.parametrize(
+        'kernel_sizes, hidden_channels, strides, paddings', [
+            ((5, 3), (12, 8), (1, 1), (2, 1)),
+        ])
+    def test_output_values_with_unequal_stride_with_padding(
+            self, hidden_channels, kernel_sizes, strides, paddings):
+        """Test output values with unequal stride and padding from CNNModule.
+
+        Args:
+            kernel_sizes (tuple[int]): Kernel sizes.
+            hidden_channels (tuple[int]): hidden channels.
+            strides (tuple[int]): strides.
+            paddings (tuple[int]): value of zero-padding.
+
+        """
+        model = CNNModule(input_var=self.input,
+                          hidden_channels=hidden_channels,
+                          kernel_sizes=kernel_sizes,
+                          strides=strides,
+                          paddings=paddings,
+                          padding_mode='zeros',
+                          hidden_nonlinearity=torch.relu,
+                          hidden_w_init=nn.init.xavier_uniform_)
+        output = model(self.input)
+
+        current_size = self.input_width
+        for (filter_size, stride, padding) in zip(kernel_sizes, strides,
+                                                  paddings):
+            # padding = float((filter_size - 1) / 2) # P = (F - 1) /2
+            current_size = int(
+                (current_size - filter_size + padding * 2) /
+                stride) + 1  # conv formula = ((W - F + 2P) / S) + 1
+        flatten_shape = current_size * current_size * hidden_channels[-1]
+
+        assert output.shape == (self.batch_size, flatten_shape)
```

Comment:
can you test the value here in addition to the shape? did the CNN produce the correct output?